### PR TITLE
Add a way to step through previously drawn frames inside testcases

### PR DESCRIPTION
**Pull request by smoogipoo**
Useful for debugging single-frame issues. Eventually the toolbar (along with drawvis in general) will be redesigned, awaiting on flyte.

Caveats: Does not work with buffered containers, because the framebuffer is shared.

---

https://streamable.com/hifv0

### CODE REVIEW

**Comment by smoogipoo** [comment]
@Tom94 Pinging you for a review to let me know if you're okay with another parameter being passed down GenerateDrawNodeSubtree. It's internal so... /shrug

File: osu.Framework/Graphics/UserInterface/BasicSliderBar.cs [hunk 1]
```diff
@@ -11,18 +12,25 @@ namespace osu.Framework.Graphics.UserInterface
     public class BasicSliderBar<T> : SliderBar<T>
         where T : struct, IComparable, IConvertible
     {
-        public Color4 Color
+        public Color4 TintColour
```
Comment (peppy) [inline osu.Framework/Graphics/UserInterface/BasicSliderBar.cs]: whyyyyyyyyyyyyyyyyyyyyyy

File: osu.Framework/Testing/TestBrowser.cs [hunk 2]
```diff
@@ -80,9 +80,13 @@ private void updateList(Assembly asm)
             leftFlowContainer.AddRange(TestTypes.Where(t => t.Assembly == asm).Select(t => new TestCaseButton(t) { Action = () => LoadTest(t) }));
         }
 
-        private BindableDouble rateBindable;
+        private readonly RateBindable rateBindable = new RateBindable();
+        private readonly AssemblyBindable assemblyBindable = new AssemblyBindable();
+        private readonly RunAllStepsBindable runAllStepsBindable = new RunAllStepsBindable();
```
Comment (peppy) [inline osu.Framework/Testing/TestBrowser.cs]: i cannot agree with using bindable to pass down one level. permission to remove such cases?

**Comment by peppy** [comment]
The assembly dropdown is broken in a weird way. Need to figure out why (may be an issue with underlying dropdown logic).

Note that it was broken before my refactor, too.

**Comment by smoogipoo** [comment]
The issue with the assembly dropdown is that the section is masking it.

**Pull request merged**